pkg/apis/config/defaults.go

```diff
@@ -53,26 +53,32 @@ const (
 	// DefaultMaxRevisionContainerConcurrency is the maximum configurable
 	// container concurrency.
 	DefaultMaxRevisionContainerConcurrency int64 = 1000
+
+	// DefaultAllowContainerConcurrencyZero is whether, by default,
+	// containerConcurrency can be set to zero (i.e. unbounded) by users.
+	DefaultAllowContainerConcurrencyZero bool = true
 )
 
 func defaultConfig() *Defaults {
 	return &Defaults{
-		RevisionTimeoutSeconds:       DefaultRevisionTimeoutSeconds,
-		MaxRevisionTimeoutSeconds:    DefaultMaxRevisionTimeoutSeconds,
-		UserContainerNameTemplate:    DefaultUserContainerName,
-		ContainerConcurrency:         DefaultContainerConcurrency,
-		ContainerConcurrencyMaxLimit: DefaultMaxRevisionContainerConcurrency,
+		RevisionTimeoutSeconds:        DefaultRevisionTimeoutSeconds,
+		MaxRevisionTimeoutSeconds:     DefaultMaxRevisionTimeoutSeconds,
+		UserContainerNameTemplate:     DefaultUserContainerName,
+		ContainerConcurrency:          DefaultContainerConcurrency,
+		ContainerConcurrencyMaxLimit:  DefaultMaxRevisionContainerConcurrency,
+		AllowContainerConcurrencyZero: DefaultAllowContainerConcurrencyZero,
 	}
 }
 
-// NewDefaultsConfigFromMap creates a Defaults from the supplied Map
+// NewDefaultsConfigFromMap creates a Defaults from the supplied Map.
 func NewDefaultsConfigFromMap(data map[string]string) (*Defaults, error) {
 	nc := defaultConfig()
 
-	// Process bool field.
+	// Process bool fields.
 	nc.EnableMultiContainer = strings.EqualFold(data["enable-multi-container"], "true")
+	nc.AllowContainerConcurrencyZero = !strings.EqualFold(data["allow-container-concurrency-zero"], "false")
```

Comment:
Can this copy the `Process bool fields.` method from the autoscaler config?

Reply:
yep 👍